firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java

```diff
@@ -326,4 +327,13 @@ public FieldIndex decodeFieldIndex(
         fieldIndex.withVersion(new SnapshotVersion(new Timestamp(updateSeconds, updateNanos)));
     return fieldIndex;
   }
+
+  public Mutation decodeMutation(byte[] blob) throws InvalidProtocolBufferException {
+    Write mutation = Write.parseFrom(blob);
+    return rpcSerializer.decodeMutation(mutation);
+  }
+
+  public byte[] encodeMutation(Mutation mutation) {
+    return rpcSerializer.encodeMutation(mutation).toByteArray();
+  }
```

Comment:
I just scrolled through the file and noticed that none of the other methods here encode and decode to and from bytes. Rather, this is done at the callsite. This actually has some benefit as this "detaches" this encoder from the underlying storage layer. While SQLite uses `byte[]`, the memory implementation uses the Proto directly. With that, is it possible to convert to bytes in the SQLite implementation of the document overlay?

Reply:
Makes sense.